DANS-KNAW/easy-deposit-api
# EASY-2222 overide stacktrace by ingest-flow with mail-to-dans message

Fixes EASY-2222 overide stacktrace by ingest-flow with mail-to-dans message

#### When applied it will
* generate mailto-links with two variants of generated references:
  * one with the UUID found with `bag-store.bag-id` in the `deposit.properties of the draft deposit`
  * one with 'DRAFT/<user>/<UUID>' as fallback if the property (or submitted copy of the draft) was not found
* [x] fill in the title and test a long one

#### Where should the reviewer @DANS-KNAW/easy start?

The template in `StateManager.mailtoDansMssage` and when the method is called 

#### How should this be manually tested?

* [x] on deasy: 
  * `easy-dtap` and `easy-app` on ui-integration branch
  * build/deploy `easy-deposit-api` and `easy-webui`
  * shut down a service like `easy-validate-dans-bag `
  * submit a deposit (with at least one file uploaded)
  * refresh the list of deposits until ingest-flow is done
* Try the `expectedURL`-s in `StateManagerSpec`, result is shown in the screenshots. 

#### Screen shots
* The link only shows up in the notes of the deposit-ui when the mouse hovers over it
* Note that the deposit-ui does not show the user-id. The user-id does appear in the the message greeting. Users might change it and/or an automated signature gets added.
* Compare the trimmed titles in the `dataset.json` snippets with the subject line and message body.
![image](https://user-images.githubusercontent.com/10553630/62530448-cdb32c00-b840-11e9-961e-6f1fbc267aa2.png)
![image](https://user-images.githubusercontent.com/10553630/62531026-e2dc8a80-b841-11e9-9813-72733f715f86.png)

https://github.com/DANS-KNAW/easy-deposit-api/blob/9ee217567ee89d979bd26c977684637d94276e7c/src/test/scala/nl.knaw.dans.easy.deposit/StateManagerSpec.scala#L132-L133
![image](https://user-images.githubusercontent.com/10553630/62529156-64cab480-b83e-11e9-9f4e-b15b0407a77c.png)

https://github.com/DANS-KNAW/easy-deposit-api/blob/9ee217567ee89d979bd26c977684637d94276e7c/src/test/scala/nl.knaw.dans.easy.deposit/StateManagerSpec.scala#L150-L156
![image](https://user-images.githubusercontent.com/10553630/62529261-9ba0ca80-b83e-11e9-8af3-29765c2958ab.png)
